src/components/StampOverlay/StampOverlay.js

```diff
@@ -50,14 +53,17 @@ class StampOverlay extends React.Component {
   componentDidUpdate(prevProps) {
     // if language changes while overlay is open we wanted update it
     const isLanChanged = this.state.language && this.props.i18n.language !== this.state.language;
-
-    if (prevProps.activeToolName !== this.props.activeToolName && this.props.activeToolName === TOOL_NAME) {
-      this.setState({ isStampSelected: false });
-      this.props.openElement('stampOverlay');
+    if (!prevProps.isOpen && this.props.isOpen) {
+      this.props.closeElements([
+        'viewControlsOverlay',
+        'menuOverlay',
+        'toolsOverlay',
+        'zoomOverlay',
+        'toolStylePopup',
+      ]);
       this.setOverlayPosition();
       this.getDefaultRubberStamps(isLanChanged);
-    } else if (this.props.isOpen && isLanChanged) {
-      this.getDefaultRubberStamps(isLanChanged);
+      this.getCustomRubberStamps(isLanChanged);
```

Comment:
I feel like we don't need to call `getDefaultRubberStamps` and `getCustomRubberStamps` every time the overlay opens. Maybe just when the language changes?

Do you think it makes sense to have these two function calls in `componentDidMount` and then in `componentDidUpdate` we only call them when the language changes? I think this can improve some performance, and is a relative change that we can do.

Reply:
Yes I think that makes sense. My original code was doing that, and later it got changed to something else. I will check language change and update accordingly